File: C/第一部分 C语言/第二章 顺序结构程序设计/第六节 顺序结构实例/sequence_case.c
// 由bhw设计
/*
第2章 第6节 顺序结构实例
题目: 输入长方形长和宽, 输出面积和周长。
思路:
1) 读取length和width。
2) 计算area与perimeter。
3) 格式化输出结果。

习题:
1) 改成输入圆半径输出面积和周长。
2) 增加输入合法性判断(必须大于0)。

答案与讲解:
1) 圆面积PI*r*r, 周长2*PI*r。
2) if (length<=0 || width<=0) 报错并退出。
*/
#include <stdio.h> // 使用scanf/printf

int main(void) { // 主函数开始
    double length = 0.0; // 定义长
    double width = 0.0; // 定义宽
    if (scanf("%lf%lf", &length, &width) != 2) { // 读取两个浮点数
        printf("输入失败\n"); // 输入失败提示
        return 1; // 异常返回
    } // 输入成功后继续
    double area = length * width; // 面积计算
    double perimeter = 2.0 * (length + width); // 周长计算
    printf("area=%.2f perimeter=%.2f\n", area, perimeter); // 输出结果
    return 0; // 正常结束
} // 主函数结束
// 由bhw设计

File: C#/第一部分 C#语言/第二章 顺序结构程序设计/第六节 顺序结构实例/sequence_case.cs
// 由bhw设计
/*
第2章 第6节 顺序结构实例

题目：输入长方形的长和宽，输出面积和周长。

一、解题思路
1) 读入 length 与 width。
2) 面积 area = length * width。
3) 周长 perimeter = 2 * (length + width)。

二、习题
1) 改成输入圆半径，输出面积和周长。
2) 增加输入合法性判断（长宽必须 > 0）。

三、答案与讲解
1) 圆面积 PI*r*r，周长 2*PI*r。
2) if (length <= 0 || width <= 0) 直接提示并返回。
*/
using System; // 控制台与基础类型

class Program // 程序类
{
    static void Main() // 入口函数
    {
        Console.Write("请输入长和宽（空格分隔）: "); // 提示输入
        string line = Console.ReadLine() ?? ""; // 读取一行
        string[] parts = line.Split(' ', StringSplitOptions.RemoveEmptyEntries); // 按空格拆分

        if (parts.Length != 2) // 校验输入项数量
        {
            Console.WriteLine("输入格式错误"); // 错误提示
            return; // 结束
        }

        if (!double.TryParse(parts[0], out double length) || !double.TryParse(parts[1], out double width)) // 双重解析
        {
            Console.WriteLine("数字格式错误"); // 解析失败提示
            return; // 结束
        }

        double area = length * width; // 计算面积
        double perimeter = 2 * (length + width); // 计算周长
        Console.WriteLine($"area={area:F2}, perimeter={perimeter:F2}"); // 格式化输出
    } // Main 结束
} // 类结束
// 由bhw设计
